src/coreclr/jit/importercalls.cpp

```diff
@@ -8540,6 +8541,17 @@ NamedIntrinsic Compiler::lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method)
                     }
                 }
             }
+            else if (strcmp(namespaceName, "Text") == 0)
+            {
+                if (strcmp(className, "UTF8EncodingSealed") == 0)
```

Comment:
For the UTF8 string interpolation, would it be better to just have a simple static `UTF8.TryGetBytes` method that would be easier for the JIT to deal with?

It feels like the JIT spends a lot of effort to reverse engineer the pattern. It is likely going to eat into inlining budgets and other limitations. How complicated the interpolated string can be before the optimizations get turned off? 

cc @stephentoub

Reply:
If it's really a problem, we could at a minimum add that as an internal method that Utf8.TryWrite would use.  I'm not super excited about adding two ways to do the exact same thing publicly, i.e. Utf8.TryGetBytes vs Encoding.UTF8.TryGetBytes, though I also don't love the prospect of having the special one we use internally in corelib and a public one available for everything else, so ideally this would only be a medium-term mitigation.

Reply:
Initially, I planned to intrinsify the public API itself `bool TryGetBytes(ReadOnlySpan<char> chars, Span<byte> bytes, out int bytesWritten)` and even made a prototype for. I then realized that Span arguments are very hard to handle in JIT because the way we do ABI preparations too early, I migth try again that.

But an api that doesn't take spans as arguments is a lot easier to work with

Reply:
Let me try to handle `TryGetBytes` itself. It is currently never inlined without Dynamic PGO so I won't regress anyone by adding a NoInlining attribute for it. (well, except PGO users)

Reply:
> > But an api that doesn't take spans as arguments is a lot easier to work with
> 
> So something like `private static bool TryGetBytes(ref char source, int sourceLength, ref byte destination, int destinationLength)`?

Yes, but let me see if Span can work too again